notebook/services/contents/fileio.py

```diff
@@ -30,6 +30,16 @@
     from base64 import encodestring as encodebytes, decodestring as decodebytes
 
 
+def replace_file(src, dst):
+    """ replace dst with src
+
+    switches between os.replace or os.rename based on python 2.7 or python 3
+    """
+    try:
+        os.replace(src, dst)
+    except:
```

Comment:
So here don't leave a catch all `except:` clause or you will catch unwanted errors like permission errors, disk full ...and so on, and so forth, so when you can, it is better to be explicit about exception you catch when possible. 

Here it's likely not possible, so you you will have to check with an `if/else ` and `sys.version_info >= (3,3)`